### DISCUSSION TITLE
Can I retrieve information about the queue to control the enqueue speed?

### DISCUSSION THREAD

**Discussion by piratf** [hibiken/asynq#748] [Q&A] [answered]
I have a client that places tasks in a queue, and there are servers responsible for processing these tasks.

I aim to maintain an optimal queue size in order to ensure stability. For instance, if I can retrieve the current queue size, I can instruct the client to wait until the size falls below certain limits.

Are there any established practices for achieving this?

**Comment by kamikazechaser** [comment, marked as answer]
Yeah, you can use https://pkg.go.dev/github.com/hibiken/asynq#Inspector
